### PR TITLE
Add caller and test name info in log entry

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:

The log wrapper for logrus written in `pkg/log/log.go` did not provide us the information regarding the current test case which is being executed and also the caller function with the line number. This is sometimes cumbersome to debug issues and zero upon the block of code causing the failure.
This PR fixes the problem by adding the test case name as part of the log formatter and also provides the caller function information along with the line from which the log function is being called.

Before this PR logs looked like this:

```
2023-02-19 11:04:31 +0000:[INFO log.go:#237]  dump kubeconfigs to file system
2023-02-19 11:04:31 +0000:[INFO log.go:#237]  Get over kubeconfig list [source-config destination-config]
2023-02-19 11:04:31 +0000:[INFO log.go:#237]  Save kubeconfig to /tmp/source-config
2023-02-19 11:04:31 +0000:[INFO log.go:#237]  Save kubeconfig to /tmp/destination-config
2023-02-19 11:04:31 +0000:[INFO log.go:#242]  Create cluster [source-cluster] in org [default]
2023-02-19 11:04:31 +0000:[INFO log.go:#237]  Source config path: /tmp/source-config
2023-02-19 11:04:31 +0000:[INFO log.go:#237]  Save cluster source-cluster kubeconfig to /tmp/source-config
2023-02-19 11:04:31 +0000:[INFO log.go:#242]  Create cluster [source-cluster] in org [default]
2023-02-19 11:04:32 +0000:[INFO log.go:#242]  Create cluster [destination-cluster] in org [default]
2023-02-19 11:04:32 +0000:[INFO log.go:#237]  Destination config path: /tmp/destination-config
2023-02-19 11:04:32 +0000:[INFO log.go:#237]  Save cluster destination-cluster kubeconfig to /tmp/destination-config
2023-02-19 11:04:32 +0000:[INFO log.go:#242]  Create cluster [destination-cluster] in org [default]
```

As we can see there is no information regarding the caller function and the line number

Now it would look like this:

```
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.dumpKubeConfigs:#3890] - dump kubeconfigs to file system
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.dumpKubeConfigs:#3896] - Get over kubeconfig list [source-config destination-config]
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.dumpKubeConfigs:#3905] - Save kubeconfig to /tmp/source-config
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.dumpKubeConfigs:#3905] - Save kubeconfig to /tmp/destination-config
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] Create cluster [source-cluster] in org [default]
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.GetSourceClusterConfigPath:#3332] - Source config path: /tmp/source-config
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.CreateSourceAndDestClusters:#2833] - Save cluster source-cluster kubeconfig to /tmp/source-config
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] Create cluster [source-cluster] in org [default]
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] Create cluster [destination-cluster] in org [default]
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.GetDestinationClusterConfigPath:#3349] - Destination config path: /tmp/destination-config
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] [tests.CreateSourceAndDestClusters:#2844] - Save cluster destination-cluster kubeconfig to /tmp/destination-config
2023-02-18 23:15:41 +0000:[INFO] [{CancelClusterBackupShare}] Create cluster [destination-cluster] in org [default]
```

Here we know that the test name is `{CancelClusterBackupShare}` and the line number is appended after the function name - `tests.GetDestinationClusterConfigPath:#3349` which will ease the debugging process.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:
Here is a full [jenkins run](https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/PX-Backup/job/system-tests/job/px-backup-system-test/120/consoleFull) where I have executed all the Px-Backup system tests

Please feel free to use this branch and test out your code/test cases to see if everything works as expected.

### CODE REVIEW

**Comment by vilasd** [comment]
@mkoppal-px this is really good PR to capture caller and line number for easy debugging.

**Review by vilasd — COMMENTED** [review]
Do these get into Aetos logs ?

**Comment by mkoppal-px** [comment]
> Do these get into Aetos logs ?

No, these will show-up only on the console log and file logs

File: pkg/log/log.go [hunk 1]
```diff
@@ -197,44 +199,72 @@ type MyFormatter struct{}
...
 
-	b.WriteString(fmt.Sprintf("%s:[%s %s:#%d]  %s\n",
-		entry.Time.Format("2006-01-02 15:04:05 -0700"), level, fileName, entry.Caller.Line,
-		entry.Message))
+	// TODO: Once we migrate to ginkgo v2, we can extract test name as follows:
+	// report := CurrentSpecReport()
+	// testName := report.ContainerHierarchyTexts[0]
+	report := CurrentGinkgoTestDescription()
+	testName := strings.Split(report.FullTestText, " ")[0]
+	if testName != "" {
```
Comment (vprabhakar-px) [inline pkg/log/log.go:221]: can we have the format from "[{CancelClusterBackupShare}] [tests.dumpKubeConfigs:#3890]" to 
[CancelClusterBackupShare:tests.dumpKubeConfigs:#3890]

**Review by vprabhakar-px — CHANGES_REQUESTED** [review]

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

File: pkg/log/log.go [hunk 2]
```diff
@@ -197,44 +199,72 @@ type MyFormatter struct{}
...
+	}
+
+	b.WriteString(writeString)
 	return b.Bytes(), nil
 }
 
 func Fatalf(format string, args ...interface{}) {
-	dash.Fatal(format, args...)
-	tpLog.Errorf(format, args...)
+	pc, _, line, _ := runtime.Caller(1)
```
Comment (Madanagopal19) [inline pkg/log/log.go:236]: @mkoppal-px , looks like these below 4 lines are repeated in all the methods in log.go file, shall we wrap together in a method named `extendedFormat` or something similar like that ? 
```  
        pc, _, line, _ := runtime.Caller(1)
	callerFuncSlice := strings.Split(runtime.FuncForPC(pc).Name(), "/")
	callerFunc := fmt.Sprintf("%s:#%d", callerFuncSlice[len(callerFuncSlice)-1], line)
	extendedFormat := fmt.Sprintf("[%s] - %s", callerFunc, format)
```

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

File: pkg/log/log.go (hunk 2 above)
Comment (lsrinivas-pure) [reply pkg/log/log.go:236]: @Madanagopal19 if we create another function, the caller function will change and it will not be representing accurately

Comment (Madanagopal19) [reply pkg/log/log.go:236]: got it.

**Review by Madanagopal19 — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

File: pkg/log/log.go (hunk 1 above)
Comment (vprabhakar-px) [reply pkg/log/log.go:221]: LGTM

**Review by vprabhakar-px — APPROVED** [review]

**Pull request merged**